pkg/operator/apiservicecontroller/apiservice_controller.go

```diff
@@ -230,3 +255,93 @@ func (c *APIServiceController) eventHandler() cache.ResourceEventHandler {
 		DeleteFunc: func(obj interface{}) { c.queue.Add(workQueueKey) },
 	}
 }
+
+// TODO: change the name
+func (c *APIServiceController) filterAPIServices() []*apiregistrationv1.APIService {
+
+	oauthOperatorPrecondition := c.oauthOperatorAvailableAndVersionMatch()
+
+	filteredApiServices := []*apiregistrationv1.APIService{}
+
+	// the c.apiServices list is authoritative
+	// it can compete with oauth-server in case of errors but it could save us on downgrade (the current coao and an old cao)
+	for _, apiService := range c.apiServices {
+		if c.isAPIServicePotentiallyManagedByExternalServer(apiService) && oauthOperatorPrecondition && c.isAPIServiceAnnotatedByExternalServer(apiService) {
+			// TODO: log that this resource is being skipped on behalf of oauth-server
+			continue
+		}
+		filteredApiServices = append(filteredApiServices, apiService)
+	}
+
+	return filteredApiServices
+}
+
+func (c *APIServiceController) oauthOperatorAvailableAndVersionMatch() bool {
+	oauthOperator, err := c.clusterOperatorLister.Get("authentication")
```

Comment:
get the authentication.operator.openshift.io resource instead and create a new status field on it that says, "I'm managing this" then backport it to 4.3 and set it to false.

Reply:
`authentication.operator.openshift.io` doesn't provide a version information, the `KEP` explicitly states that we should compare the versions - https://github.com/deads2k/openshift-enhancements/blob/011cb767fb0e769f60f73af8dfb501847e458161/enhancements/authentication/separate-oauth-resources.md

Reply:
Yeah, turns out that doesn't work when compared against reality :(  .  Versions don't always exist.  We need a non-version dependent mechanism. Sorry, I didn't think about it until I saw it.